src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -57,8 +61,47 @@
     @ReceiveEvent(components = {SetTeamOnActivateComponent.class})
     public void onActivate(ActivateEvent event, EntityRef entity) {
         EntityRef player = event.getInstigator();
-        String team = setPlayerTeamToTeleporterTeam(player, entity);
-        handlePlayerTeleport(player, team);
+        if (isProperTeamSize(entity, player)) {
+            String team = setPlayerTeamToTeleporterTeam(player, entity);
+            handlePlayerTeleport(player, team);
+        }
+    }
+
+    private boolean isProperTeamSize(EntityRef teleporter, EntityRef player) {
+        setGameEntity();
+        int oppositeTeamCount = 0;
+        int teleporterTeamCount = 0;
+        int maxTeamSizeDifference = gameEntity.getComponent(LASConfigComponent.class).maxTeamSizeDifference;
+        String teleporterTeam = teleporter.getComponent(LASTeamComponent.class).team;
+        Iterable<EntityRef> characters = entityManager.getEntitiesWith(PlayerCharacterComponent.class,
+                LASTeamComponent.class);
+
+        for (EntityRef character : characters) {
+            String otherPlayerTeam = character.getComponent(LASTeamComponent.class).team;
+            if (teleporterTeam.equals(otherPlayerTeam)) {
+                teleporterTeamCount++;
+            } else if (!otherPlayerTeam.equals(LASUtils.WHITE_TEAM)) {
+                oppositeTeamCount++;
+            }
+        }
+        if (teleporterTeamCount - oppositeTeamCount < maxTeamSizeDifference) {
+            return true;
+        } else {
+            if (maxTeamSizeDifference == 1) {
+                player.getOwner().send(new ChatMessageEvent("The " + teleporterTeam + " team has " + maxTeamSizeDifference + " player more, so please join the "
+                        + LASUtils.getOppositionTeam(teleporterTeam) + " team.", EntityRef.NULL));
+            } else {
+                player.getOwner().send(new ChatMessageEvent("The " + teleporterTeam + " team has " + maxTeamSizeDifference + " players more, so please join the "
+                        + LASUtils.getOppositionTeam(teleporterTeam) + " team.", EntityRef.NULL));
+            }
+            return false;
+        }
+    }
+
+    private void setGameEntity() {
+        if (entityManager.getCountOfEntitiesWith(LASConfigComponent.class) == 0) {
+            gameEntity = entityManager.create("LightAndShadowResources:gameEntity");
+        }
```

Comment:
You've technically build _lazy evaluation_ for the game entity - it is only created on first access. Is that something you did on purpose?

What happens if you load a game of LightAndShadow and there is already a game state entity? 

I would have expected that we reuse the entity if there is one (and probably throw and error if there are two or more, for whatever reason).

```suggestion
		 int numberOfGameStateEntities = entityManager.getCountOfEntitiesWith(LASConfigComponent.class);	
        if (numberOfGameStateEntities == 0) {
            gameEntity = entityManager.create("LightAndShadowResources:gameEntity");
        } else if (numberOfGameStateEntities == 1) {
        	 gameEntity = entityManager.getEntitiesWith(...).get(0) // get the first entity
        } else {
        	// more than two game state entities
        	// log an error or warning and crash the game mode (can we crash back to main window?)
        }
```

Reply:
I didn't consider the case of using entities when creating a saved game, I initially did use an entity if it already existed but as the entity wasn't created anywhere else currently, I removed it. Will test this out in saved games and make the change.